src/rgw/rgw_tag.h

```diff
@@ -0,0 +1,39 @@
+#ifndef RGW_TAG_H
+#define RGW_TAG_H
+
+#include <map>
+#include <string>
+#include <include/types.h>
+
+#include "rgw_common.h"
+
+class RGWObjTags
+{
+ protected:
+  std::map <std::string, std::string> tags;
```

Comment:
flat_map seems like a good fit here-- OTOH, it can also be done later. Do we know how we're actually using this, @theanalyst? For example, are there lots of small transactions on "tags", or does it just tend to grow?